core/range-controller.js

```diff
@@ -773,7 +773,13 @@ var RangeController = exports.RangeController = Montage.specialize( /** @lends R
                 // remove all values from the selection that were removed (but
                 // not added back)
                 minus.deleteEach(plus, equals);
-                if (this.selection) {
+
+                if (this.selection.length) {
+                    // ensure selection always has content
+                    if (this.content.length && this.avoidsEmptySelection && !this.multiSelect) {
+                        // selection can't contain previous content value as content already changed
+                        this.selection.add(this.content[this.content.length - 1]);
+                    }
```

Comment:
An example would be a table view bound to data fetched from a server. If the user clicks on a table cell to select it, then server deletes that cell's data, which causes the cell to be removed, what should selection include? `avoidsEmptySelection` should guarantee that selection will never be empty, because there may be code depending on its content.